src/allmydata/node.py

```diff
@@ -146,7 +146,7 @@ def read_config(basedir, portnumfile, generated_files=[], _valid_config_sections
     except EnvironmentError:
         if os.path.exists(config_fname):
             raise
-        configutil.validate_config(config_fname, parser, _valid_config_sections())
+    configutil.validate_config(config_fname, parser, _valid_config_sections())
     return _Config(parser, portnumfile, config_fname)
```

Comment:
Hmm.  The behavior when config fails to validate is not very nice now:
```
$ tahoe run foo/
daemonizing in '/tmp/foo'
running node in '/tmp/foo'
2018-05-23T08:58:50-0400 [twisted.scripts._twistd_unix.UnixAppLogger#info] twistd 18.4.0rc1 (/home/exarkun/Environments/tahoe-lafs/bin/python 2.7.14) starting up.
2018-05-23T08:58:50-0400 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
2018-05-23T08:58:50-0400 [-] Unhandled Error
        Traceback (most recent call last):
          File "/home/exarkun/Environments/tahoe-lafs/local/lib/python2.7/site-packages/twisted/internet/base.py", line 428, in fireEvent
            DeferredList(beforeResults).addCallback(self._continueFiring)
          File "/home/exarkun/Environments/tahoe-lafs/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 321, in addCallback
            callbackKeywords=kw)
          File "/home/exarkun/Environments/tahoe-lafs/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 310, in addCallbacks
            self._runCallbacks()
          File "/home/exarkun/Environments/tahoe-lafs/local/lib/python2.7/site-packages/twisted/internet/defer.py", line 653, in _runCallbacks
            current.result = callback(current.result, *args, **kw)
        --- <exception caught here> ---
          File "/home/exarkun/Environments/tahoe-lafs/local/lib/python2.7/site-packages/twisted/internet/base.py", line 441, in _continueFiring
            callable(*args, **kwargs)
          File "/home/exarkun/Work/LeastAuthority/python/tahoe-lafs/src/allmydata/scripts/tahoe_daemonize.py", line 128, in start
            srv = service_factory()
          File "/home/exarkun/Work/LeastAuthority/python/tahoe-lafs/src/allmydata/scripts/tahoe_daemonize.py", line 117, in <lambda>
            u"client": lambda: namedAny("allmydata.client.create_client")(self.basedir),
          File "/home/exarkun/Work/LeastAuthority/python/tahoe-lafs/src/allmydata/client.py", line 160, in create_client
            config = read_config(basedir, u"client.port", _valid_config_sections=_valid_config_sections)
          File "/home/exarkun/Work/LeastAuthority/python/tahoe-lafs/src/allmydata/node.py", line 149, in read_config
            configutil.validate_config(config_fname, parser, _valid_config_sections())
          File "/home/exarkun/Work/LeastAuthority/python/tahoe-lafs/src/allmydata/util/configutil.py", line 60, in validate_config
            option=option,
        allmydata.util.configutil.UnknownConfigError: '/tmp/foo/tahoe.cfg' section [storage] contains unknown option 'storage_idr'
```

Here it hangs ("runs" I suppose but not doing anything useful) until Ctrl-C.  I would guess this wasn't always so... I wonder if some refactoring caused this?

This is not a blocker for this fix but maybe it is a blocker for the release.  I see that 1.12.1 exits on this exception.

Reply:
Yeah, that should be fixed. Thanks, I'll take a look!